hibernate-core/src/main/java/org/hibernate/query/sqm/internal/SqmCriteriaNodeBuilder.java

```diff
@@ -1229,6 +1229,10 @@ public <T> List<? extends SqmExpression<T>> literals(List<T> values) {
 
 	@Override
 	public <T> SqmExpression<T> nullLiteral(Class<T> resultClass) {
+		if ( resultClass.isEnum() ) {
```

Comment:
I think you break CriteriaQuery with multiselect and nullLiteral for enum types this way because we have a constructor that expects an enum and it gets an Object instead: 
https://github.com/hibernate/hibernate-orm/blob/main/hibernate-core/src/main/java/org/hibernate/sql/results/graph/instantiation/internal/DynamicInstantiationResultImpl.java#L158
EnumJavaType vs ObjectJavaType

Reply:
Please create a new bug report in the [issue tracker](https://hibernate.atlassian.net) with a [test case](https://github.com/hibernate/hibernate-test-case-templates/blob/master/orm/hibernate-orm-6/src/test/java/org/hibernate/bugs/JPAUnitTestCase.java) that reproduces the problem you're experiencing if you want us to look into it.

Reply:
I'm not sure if I should report it or not. It works on the main branch and it fails for 6.2.7 (currently included in spring boot dependencies - 3.1.3). Test in `org.hibernate.orm.test.jpa.criteria.nulliteral.NullLiteralExpression`

```
	@Test
	public void testEnumNullLiteralMultiselect() {
		doInJPA( this::entityManagerFactory, entityManager -> {
			final CriteriaBuilder builder = entityManager.getCriteriaBuilder();
			final CriteriaQuery<Person> criteriaQuery = builder.createQuery( Person.class );
			Root<Person> person = criteriaQuery.from( Person.class );
			criteriaQuery.multiselect(
					person.get(Person_.id),
					person.get(Person_.subject),
					builder.nullLiteral(EyeColor.class)
			);
			entityManager.createQuery( criteriaQuery ).getResultList();
		} );
	}
```
with an additional constructor in the Person class:
```
	public Person(long id, Subject subject, EyeColor eyeColor) {
		this.id = id;
		this.subject = subject;
		this.eyeColor = eyeColor;
	}
```

Reply:
This change didn't break that test. It wasn't working before (since 6.0 most likely). It would break proper behavior if you worked around this problem by contributing basictype for required enums.